[MTIAGraph][Pytorch] Add the graph_pool_handle api

andyanwang opened the pull request:
Summary:
Add the `torch.mtia.graph_pool_handle` API as the counterpart of `torch.cuda.graph_pool_handle`, which is used in vllm, e.g.
https://www.internalfb.com/code/fbsource/[f6d024bd45964d71810cbe1ed859f132f7f734cd]/fbcode/vllm/trunk/vllm/compilation/cuda_graph.py?lines=170

Test Plan:
```
buck2 run mtia/host_runtime/torch_mtia/tests:test_mtia_graph_py -- -r test_graph_pool_handle
```

Differential Revision: D88059625




cc @egienvalue

patrick-toulme reviewed (approved):
LGTM. Please explain the cast part though

andyanwang committed:
[MTIAGraph][Pytorch] Add the graph_pool_handle api (#169283)

Summary:

Add the `torch.mtia.graph_pool_handle` API as the counterpart of `torch.cuda.graph_pool_handle`, which is used in vllm, e.g.
https://www.internalfb.com/code/fbsource/[f6d024bd45964d71810cbe1ed859f132f7f734cd]/fbcode/vllm/trunk/vllm/compilation/cuda_graph.py?lines=170

Test Plan:
```
buck2 run mtia/host_runtime/torch_mtia/tests:test_mtia_graph_py -- -r test_graph_pool_handle
```

Differential Revision: D88059625
